client.go

```diff
@@ -101,6 +101,13 @@ func (c *Client) ForwardGeocode(ctx context.Context, req *ForwardGeocodeRequest)
 	return forwardGeocode(ctx, c, req)
 }
 
+func (c *Client) Directions(ctx context.Context, req *DirectionsRequest) (*DirectionsResponse, error) {
+	if err := c.checkRateLimit(Matrix); err != nil {
```

Comment:
I think we can setup a new rate limit for `Directions` here since it may need different limits than matrix endpoints?